### ISSUE TITLE
Localization

### ISSUE THREAD

**Issue by serwizz** [sissaschool/xmlschema#292]
Hi!

Thank you for your great job with xmlschema! I use xmlschema for xml validation but I need to localize error messages. Unfortunately right now it is impossible. What do you think if I will add gettext support? I can make some pull requests with base localization code, e.g. python gettext initialization and wrap error messages.

Something like this:

**init.py**
```python
folder = pathlib.Path(__file__).parent.resolve()
translation = gettext.translation(
    'xmlschema',
    pathlib.Path.joinpath(folder, 'locale'),
    fallback=True,
)
translation.install()
```

**validators/exceptions.py**
```python
reason = _("Unexpected child with tag %r at position %d.") % (child_tag, index + 1)
...
reason += _(" Tag %r expected.") % expected_tags[0]
```
Of course existing English messages will remain the same.

Thanks.

**Comment by brunato** [comment]
Hi,

this can be very useful, maybe gettext/ngettext could be needed in other library modules (not only in **validators/exceptions.py**) where error messages are defined.

The **init.py** part should be left at the application level I suppose. Isn't it?

Thanks

**Comment by serwizz** [comment]
Hi, 
I'm added some code in first draft by #293 
Just for example and for further discussion.

Thanks

**Comment by brunato** [comment]
Hi,
i added some comments on your PR.

I don't want to install translation into package `__init__.py` because someone might not want to activate translations at all or might want to use a custom translation. I propose to add an utility function for activating translations (into helpers module or into a new module).

Thanks

**Comment by serwizz** [comment]
Hi,
I agree it is not the best solution. I will make some wrapper for translation function or so. 

Thanks

**Comment by brunato** [comment]
Hi,
in the version v1.11.0 I have included translations for parsing/validation error messages. 
For activating the translation you can use:

    import xmlschema
    xmlschema.translation.activate()   # or xmlschema.translation.activate(languages=['ru'])

Translation can be also deactivated:

    xmlschema.translation.deactivate()

Translations are not installed in builtins for default (provide `activate(install=True)`).

At schema level I've removed the proposed options because currently are not manageable at instance level with coherence.

I have built a new xmlschema.POT template, that can be considered completed (I don't want to extend it to error
messages not related with validation/parsing).

The version v1.11.0 includes a full Italian translation and a partial Russian translation derived from your PR. Complete 
with a full Russian translation with another PR and then i will publish a new release.

**Comment by brunato** [comment]
Hi @serwizz,
any progress for including a full Russian translation based on v1.11.0 release?

I will publish soon a new release with fixes, and I'm uncertain that keeping heavily incomplete translations in the published package is a good thing (in any case one can create and use it's own custom translation locally for his installations).

Thank you

**Comment by serwizz** [comment]
Hi, I'm working on it, I'll try to finish it in the next few days. Sorry for delay.

**Comment by serwizz** [comment]
Hi,
I create PR with russian locale #303

**Comment by brunato** [comment]
I have merged the PR #303 but I report you that one message has not been translated:

```
#: xmlschema/validators/elements.py:365
#, python-format
msgid "head element %r can't be substituted by an element that has a derivation of its type"
msgstr "head element %r can't be substituted by an element that has a derivation of its type"
```

If you post me a translation for this i will modify for you, otherwise you will make another PR in the future.

Thank you

**Comment by serwizz** [comment]
Hi,
I fixed the untranslated message and updated russian .po file from the .pot file in #304.

Thank you

**Comment by brunato** [comment]
Russian translation included in release v1.11.2.

Best
